### ISSUE TITLE
Style/ModuleMemberExistenceCheck Autocorrection Potentially Unsafe for const_defined?

### ISSUE THREAD

**Issue by drcapulet** [rubocop/rubocop#15040]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/ModuleMemberExistenceCheck

### Steps to reproduce the problem

1. Create a file named example.rb with the following contents:
  ```ruby
# frozen_string_literal: true

module MyClass
    class MySubClass
    end
end

puts MyClass.constants.include?(:String)
```
2. Run `rubocop -a example.rb`

### Expected behavior

`const_defined?` is not a 1-1 replacement for `constants.include?` - the former is checking if a reference is resolvable inside the context of a class / module, whereas [`.constants`](https://ruby-doc.org/3.4.1/Module.html#method-i-constants) returns the constants inside of the caller. Before autocorrect, `ruby example.rb` would print `false` but after prints `true`.

### Actual behavior

```
For ~/rubocop-test: configuration from ~/rubocop-test/.rubocop.yml
Default configuration from ~/.asdf/installs/ruby/3.4.8/lib/ruby/gems/3.4.0/gems/rubocop-1.85.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ~/rubocop-test/example.rb
Loading cache from ~/.cache/rubocop_cache/ccedc180d2af4eee4376a7e20bdf0a49bba19cdf/6d7a3b621ca1730e04accd938619e4bdab66cfb1/616aee62c1c9a74a36ca3ce09abf0cf538fbf6df
W

Offenses:

example.rb:4:3: W: Lint/EmptyClass: Empty class detected.
  class MySubClass ...
  ^^^^^^^^^^^^^^^^
example.rb:8:14: C: [Corrected] Style/ModuleMemberExistenceCheck: Use const_defined?(:String) instead.
puts MyClass.constants.include?(:String)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 1 offense corrected
Finished in 0.0636 seconds
```

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
```

### RuboCop version

```shell
1.85.1 (using Parser 3.3.10.1, Prism 1.9.0, rubocop-ast 1.49.0, analyzing as Ruby 3.4, running on ruby 3.4.8) [arm64-darwin25]
```
